types/utils.go

```diff
@@ -133,3 +134,10 @@ func ParseLengthPrefixedBytes(key []byte, startIndex int, sliceLength int) ([]by
 
 	return byteSlice, endIndex
 }
+
+// LogDeferred logs an error in a deferred function call if the returned error is non-nil.
+func LogDeferred(logger log.Logger, f func() error) {
```

Comment:
Why cannot it just take an error as an argument. What is the use case to have a function and then call it in LogDeferred. It adds imo unnecessary boilerplate in the function call.